src/utils.jl

```diff
@@ -25,6 +25,27 @@ end
 """
 $(TYPEDSIGNATURES)
 
+Remove linearly dependent rows in `A` deduced by QR decomposition. After QR decomposition, 
+all elements are rounded to `digits` after the decimal. Rows of all zeros
+(absolute value of the each element in row ≤ `atol`) are removed.
+"""
+function _remove_lin_dep_rows_QR(A; atol = 1e-8)
+    R = qr(Array(A)).R
+
+    idxs = Int[]
+    for i = 1:size(R,1)
+        if !all(abs.(R[i,:]) .<=atol) #remove rows of all zero
```

Comment:
```suggestion
        if exists(v -> abs(v) > atol,  R[i,:]) #remove rows of all zero
```
avoids allocations

Reply:
This is just to speed it up right?

Reply:
yes absolutely. `exists(something)` also reads a bit more natural than `!any(something)`

Reply:
great thanks :)

Reply:
nice, did not know about `exists`